backends/ent/store.go

```diff
@@ -42,436 +44,401 @@ func (backend *Backend) Store(doc *sbom.Document, opts *storage.StoreOptions) er
 		}
 	}
 
-	if _, ok := opts.BackendOptions.(*BackendOptions); !ok {
+	backendOpts, ok := opts.BackendOptions.(*BackendOptions)
+	if !ok {
 		return fmt.Errorf("%w", errInvalidEntOptions)
 	}
 
-	tx, err := backend.client.Tx(backend.ctx)
-	if err != nil {
-		return fmt.Errorf("creating transactional client: %w", err)
-	}
-
-	backend.ctx = ent.NewTxContext(backend.ctx, tx)
-
-	if err := tx.Document.Create().
-		SetID(doc.Metadata.Id).
-		OnConflict().
-		Ignore().
-		Exec(backend.ctx); err != nil && !ent.IsConstraintError(err) {
-		return rollback(tx, fmt.Errorf("ent.Document: %w", err))
-	}
-
-	if err := backend.saveMetadata(doc.Metadata); err != nil {
-		return rollback(tx, err)
-	}
+	// Append annotations from opts parameter with any previously set on the backend.
+	backend.Options.Annotations = append(backend.Options.Annotations, backendOpts.Annotations...)
 
-	if err := backend.saveNodeList(doc.NodeList); err != nil {
-		return rollback(tx, err)
-	}
-
-	if err := tx.Commit(); err != nil {
-		return rollback(tx, err)
+	// Set each annotation's document ID if not specified.
+	for _, a := range backend.Options.Annotations {
+		if a.DocumentID == "" {
+			a.DocumentID = doc.Metadata.Id
+		}
 	}
 
-	return nil
+	return backend.withTx(
+		func(tx *ent.Tx) error {
+			return tx.Document.Create().
+				SetID(doc.Metadata.Id).
+				OnConflict().
+				Ignore().
+				Exec(backend.ctx)
+		},
+		backend.saveAnnotations(backend.Options.Annotations...),
```

Comment:
Don't need to pass annotations parameter since they are accessible from the backend receiver